specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/compute.json

```diff
@@ -3463,57 +3469,515 @@
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/restorePointCollections": {
+      "get": {
+        "tags": [
+          "RestorePointCollections"
+        ],
+        "operationId": "RestorePointCollections_List",
+        "description": "Gets the list of restore point collections in a resource group.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/RestorePointCollectionListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/restorePointCollections": {
+      "get": {
+        "tags": [
+          "RestorePointCollections"
+        ],
+        "operationId": "RestorePointCollections_ListAll",
+        "description": "Gets the list of restore point collections in the subscription. Use nextLink property in the response to get the next page of restore point collections. Do this till nextLink is not null to fetch all the restore point collections.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/RestorePointCollectionListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/restorePointCollections/{restorePointCollectionName}/restorePoints/{restorePointName}": {
```

Comment:
>restorePoints [](http://example.com/codeflow?start=153&length=13)

You would also need a list call for /restorePoints under a collection. See [https://armwiki.azurewebsites.net/api_contracts/guidelines/policy.html#plcy008-support-a-collection-get-and-individual-get-for-child-resource-types-even-singletons](https://armwiki.azurewebsites.net/api_contracts/guidelines/policy.html#plcy008-support-a-collection-get-and-individual-get-for-child-resource-types-even-singletons)

Reply:
Yes, I have that.
To list all restore points in a restore point collection:
GET /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/restorePointCollections/{restorePointCollectionName} with (OData $expand) $expand=restorePoints
The enum is:

"name": "$expand",
--
"in": "query",
"required": false,
"type": "string",
"description": "The   expand expression to apply on the operation. If expand=restorePoints, server   will return all contained restore points in the restorePointCollection.",
"enum": [
"restorePoints"
],
"x-ms-enum": {
"name": "RestorePointCollectionExpandOptions",
"modelAsString": true
}

Reply:
Collection GET should follow the standard pattern ARM supports.

GET [https://management.azure.com/{rootScope}/providers/{namespace}/{typeA}/{nameA}/{typeB](https://management.azure.com/{rootScope}/providers/{namespace}/{typeA}/{nameA}/{typeB)}

[https://armwiki.azurewebsites.net/introduction/concepts/resourceids.html#collection-get-get-httpsmanagementazurecomresourceid_collection](https://armwiki.azurewebsites.net/introduction/concepts/resourceids.html#collection-get-get-httpsmanagementazurecomresourceid_collection)

Using $expand will prevent platform feature like Azure policy working properly.

Reply:
1. $expand=restorePoints is already being used by existing CRP customers, so I cannot remove $expand=restorePoints from our code base.
2. We are happy to add Collection GET using this pattern: GET https://management.azure.com/{rootScope}/providers/{namespace}/{typeA}/{nameA}/{typeB}. Is it okay if we add support for that endpoint in a future Swagger release? It will take some time to add that endpoint, and I don't want to delay the release of 2021-03-01 swagger any longer.

Reply:
Yes, please add support for that. Until then it may causes CSS cases if users have policies targeting this type.

Reply:
@roipmsft , please address this comment.